pkg/kp/rollstore/consul_store.go

```diff
@@ -410,6 +410,12 @@ func (s consulStore) CreateRollingUpdateFromOneMaybeExistingWithLabelSelector(
 			return roll_fields.Update{}, err
 		}
 
+		// Copy the new RC labels to the old RC as well
```

Comment:
you should do this after `cleanupOldRC` func is set so that the label set failure cleans up the RC